cookiecutter-xblock/{{cookiecutter.repo_name}}/{{cookiecutter.package_name}}/conf/locale/en/text.po

```diff
@@ -0,0 +1 @@
+django.po
```

Comment:
@OmarIthawi we have an issue here. Is it good enough to add the link in the template like this? or should we dig deeper to change the dependency of XBlocks on `text` domain and change it to `django` ?

Reply:
I keep forgetting this @shadinaif, so it seems that you do too :)

In XBlocks we plan to store no po/mo files at all. The LMS will completely manage the translation e.g. `edx-platform:conf/plugins-locale/drag-and-drop/conf/locale/django.po`.

So this link is not needed and could be harmful.

Reply:
@shadinaif In case I didn't make myself clear, let's keep `text.po` and textjs.po` as-is because those files will be irrelevant to all of the OEP-58 work and will eventually be removed from all XBlocks.

Reply:
@OmarIthawi thank you. What about developers creating new XBlocks? we've removed the part that converts the partial file into `text.po`. Should we just add a comment to clarify that?

or maybe better: add an [option](https://github.com/openedx/edx-cookiecutters/blob/f6e6a2d0fca3d925b94aba76610db9a51aa163cd/cookiecutter-xblock/cookiecutter.json#L11) in the cookiecutter creation to ask if the package is to be integrated with atlas, with default `No` (because most usage would be from the community)

Reply:
@shadinaif if you must implement choices I recommend the following two options:

 1- Support both Atlas OEP-58 translation mechanism and the old way (like most XBlocks have been converted). _The default choice_.
 2- Support only Atlas and OEP-58. An optional but acceptable choice.
 
In my opinion Option 1 should be the only one available. After OEP-58 is finalized, we can move to Option 2 removing Option 1 altogether unless someone really wants to.

Reply:
@shadinaif I thought about it again and I think we need the following symlinks:

 - `$ symlink -s text.po django.po`

Which is the opposite of what you've done here. 

> `make extract_translations` that produces `text.po` and `textjs.po` and `make extract_translations` that produces `django.po` and `djangojs.po`.

Sorry for the conflicting recommendations but I think this will save us from having `make extract_translations` supporting two modes.

Please let me know if it's unclear or you have any questions.

Reply:
Thank you @OmarIthawi , the case is confusing! the problem with the opposite is that `openedx-tranlsations` will fail when it runs `git add` when `django.po` is the link. So, looks like we now have three options:

**Option1**: (the one I like): add an [option](https://github.com/openedx/edx-cookiecutters/blob/f6e6a2d0fca3d925b94aba76610db9a51aa163cd/cookiecutter-xblock/cookiecutter.json#L11) in the cookiecutter creation to ask if the package is to be integrated with atlas, with default No (because most usage would be from the community). This supports two modes when creating the XBlock. U2 and partner developers who create XBlocks integrated with Atlas should simply be aware of the option
 * Default (No): for the community, the translation is hosted somewhere else. `make extract_translation` command will look like this:
 `cd $(PACKAGE_NAME) && i18n_tool extract --merge-js --combine-partial --text-domain`
(it means we add a new option `--text-domain` to `i18n_tools` to rename `django.po` to `text.po` on extraction)
the extracted file name will be `text.po`
* Support atlas (Yes): for atlas-supported XBlocks. No links are added too because `text.po` will not be used anyway. `make extract_translation` command will look like the code we already have here:
 `cd $(PACKAGE_NAME) && i18n_tool extract --merge-js --combine-partial`
the extracted file name will be `django.po`

**Option2**: add the link `-s text.po django.po` and update `openedx-translations` to deal with it. (complexity added to the script)

**Option3**: (simple but lazy from our side, we put the burden on others). always add the link `-s django.po text.po` and let developers remove it manually when they know that the XBlock supports atlas

What do you think?

Reply:
@shadinaif Atlas is not for 2U although they'll be using Atlas for sure. OEP-58 is meant to be used for the wider community of Open edX.

Atlas is for the whole community, therefore we should make it enabled by default because the in-repo committed pofiles will be deprecated and removed. Therefore an option to enable Atlas doesn't make sense.


> Thank you @OmarIthawi , the case is confusing! the problem with the opposite is that `openedx-tranlsations` will fail when it runs `git add` when `django.po` is the link. So, looks like we now have three options:

Yeah, you're right. We're dealing with conflicting requirements here.

Both Option 2 and Option 3 are good.

Please make `atlas` compatibility mandatory in the cookie-cutter. Anyway, advance XBlock users can remove it of they really want to.

Reply:
> @shadinaif Atlas is not for 2U although they'll be using Atlas for sure.

this is the main point I was missing! thank you @OmarIthawi 

I'll go with Option2 then 👍🏼

Reply:
@OmarIthawi  Update on Option2:
~~add the link -s text.po django.po and~~ update openedx-translations to deal with `text.po` and `textjs.po` files when used by XBlocks. No links are added to the cookiecutter

Reply:
Thanks! @shadinaif please remove the link from this PR.